common/flatpak-dir.c

```diff
@@ -6990,6 +7004,19 @@ flatpak_dir_deploy (FlatpakDir          *self,
   if (application_runtime)
     g_variant_builder_add (&metadata_builder, "{s@v}", "runtime",
                            g_variant_new_variant (g_variant_new_string (application_runtime)));
+  if (old_deploy_data)
+    {
+      g_autofree const char **previous_ids = flatpak_deploy_data_get_previous_ids (old_deploy_data);
+      g_variant_builder_add (&metadata_builder, "{s@v}", "previous-ids",
+                             g_variant_new_variant (g_variant_new_strv ((const char * const *) previous_ids, -1)));
+    }
+  else
+    {
+      // hard coded list of dummy ids
+      const char *previous_ids[] = {"com.dummy.com", "com1.dummy1.com1", "com2.dummy2.com2", NULL};
```

Comment:
What’s the reasoning behind putting in a hard-coded list of dummy IDs?

Reply:
Sorry, I missed this comment.

This is just for testing. This should be already come in the deploy file of the app in the future, I believe. Just that, I shove it directly.

So, if you are testing this PR, and rename one of `~/.var/flatpak/app/any.exisiting.app` to one of these hard-coded IDs and re-run `flatpak run any.exisitng.app`, you should see the renaming and symlinks taking place as intended by this PR.